### PR TITLE
sundials 5.3.0

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by chenrui333** [comment]
<details>
<summary>audit failure log</summary>

```
 [31m==>[0m [1m[31mFAILED[0m[0m
 [31mError:[0m 1 problem in 1 formula detected
 sundials:
   * Files were found with references to the Homebrew shims directory.
     The offending files are:
       examples/sunnonlinsol/newton/CMakeLists.txt
       examples/sunnonlinsol/newton/Makefile
       examples/sunnonlinsol/fixedpoint/CMakeLists.txt
       examples/sunnonlinsol/fixedpoint/Makefile
       examples/sunlinsol/pcg/serial/CMakeLists.txt
       examples/sunlinsol/pcg/serial/Makefile
       examples/sunlinsol/band/CMakeLists.txt
       examples/sunlinsol/band/Makefile
       examples/sunlinsol/spbcgs/parallel/Makefile
       examples/sunlinsol/spbcgs/serial/CMakeLists.txt
       examples/sunlinsol/spbcgs/serial/Makefile
       examples/sunlinsol/dense/CMakeLists.txt
       examples/sunlinsol/dense/Makefile
       examples/sunlinsol/spfgmr/parallel/Makefile
       examples/sunlinsol/spfgmr/serial/CMakeLists.txt
       examples/sunlinsol/spfgmr/serial/Makefile
       examples/sunlinsol/sptfqmr/parallel/Makefile
       examples/sunlinsol/sptfqmr/serial/CMakeLists.txt
       examples/sunlinsol/sptfqmr/serial/Makefile
       examples/sunlinsol/spgmr/parallel/Makefile
       examples/sunlinsol/spgmr/serial/CMakeLists.txt
       examples/sunlinsol/spgmr/serial/Makefile
       examples/sunlinsol/lapackband/CMakeLists.txt
       examples/sunlinsol/lapackband/Makefile
       examples/sunlinsol/klu/CMakeLists.txt
       examples/sunlinsol/klu/Makefile
       examples/sunlinsol/lapackdense/CMakeLists.txt
       examples/sunlinsol/lapackdense/Makefile
       examples/cvode/parallel/Makefile
       examples/cvode/C_mpimanyvector/Makefile
       examples/cvode/serial/CMakeLists.txt
       examples/cvode/serial/Makefile
       examples/sunmatrix/band/CMakeLists.txt
       examples/sunmatrix/band/Makefile
       examples/sunmatrix/dense/CMakeLists.txt
       examples/sunmatrix/dense/Makefile
       examples/sunmatrix/sparse/CMakeLists.txt
       examples/sunmatrix/sparse/Makefile
       examples/ida/parallel/Makefile
       examples/ida/serial/CMakeLists.txt
       examples/ida/serial/Makefile
       examples/arkode/C_parallel/Makefile
       examples/arkode/C_manyvector/Makefile
       examples/arkode/C_serial/CMakeLists.txt
       examples/arkode/C_serial/Makefile
       examples/idas/parallel/Makefile
       examples/idas/serial/CMakeLists.txt
       examples/idas/serial/Makefile
       examples/kinsol/parallel/Makefile
       examples/kinsol/serial/CMakeLists.txt
       examples/kinsol/serial/Makefile
       examples/nvector/parallel/Makefile
       examples/nvector/mpimanyvector/Makefile
       examples/nvector/manyvector/CMakeLists.txt
       examples/nvector/manyvector/Makefile
       examples/nvector/mpiplusx/Makefile
       examples/nvector/serial/CMakeLists.txt
       examples/nvector/serial/Makefile
       examples/cvodes/parallel/Makefile
       examples/cvodes/serial/CMakeLists.txt
       examples/cvodes/serial/Makefile
```

</details>

**Comment by alebcay** [comment]
We only use the `examples/nvector` folder for testing right? We can probably remove the rest to reduce the amount of files that need to have shim references removed.

**Comment by chenrui333** [comment]
Should be good now (cannot remove the folders, as by default, the build config needs those example folders)

**Review by issyl0 — APPROVED** [review]

File: Formula/sundials.rb [hunk 1]
```diff
@@ -24,7 +23,8 @@ class Sundials < Formula
   def install
     blas = "-L#{Formula["openblas"].opt_lib} -lopenblas"
     args = std_cmake_args + %W[
-      -DCMAKE_C_COMPILER=#{ENV["CC"]}
+      -DCMAKE_C_COMPILER=/usr/bin/clang
+      -DCMAKE_CXX_COMPILER=/usr/bin/clang++
```
Comment (Bo98) [inline Formula/sundials.rb:27]: This bypasses the shims entirely during build which isn't what we want.

I argue that the makefiles should not be installed in the first place. They are generated build files - not reusable files for distribution.

**Pull request closed without merging**